libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Rotoye.cpp

```diff
@@ -0,0 +1,20 @@
+#include "AP_BattMonitor_SMBus_Rotoye.h"
+
+#include <AP_HAL/AP_HAL.h>
+
+// Specific to Rotoye Batmon
+#define BATTMONITOR_SMBUS_TEMP_EXT 0x07
+
+bool AP_BattMonitor_SMBus_Rotoye::read_temp(void) {
+    
+    uint16_t t_int, t_ext, t;
+    if (read_word(BATTMONITOR_SMBUS_TEMP, t_int) && 
```

Comment:
This doesn't look right; if you only read one you should return that, surely?

Reply:
In the case of the Rotoye Batmon, both values will always be sent over SMBus. If an external thermistor is not in use, a zero value will be sent to indicate that.

Reply:
That sort of thing should be a comment in the code.  In fact, pretty much exactly what you said :-)